<?php

namespace Test\Ease;

use Ease\Exception;

/**
 * Generated by PHPUnit_SkeletonGenerator on 2016-12-29 at 21:56:53.
 */
class ExceptionTest extends \PHPUnit_Framework_TestCase
{
    /**
     * @var Exception
     */
    protected $object;

    /**
     * Sets up the fixture, for example, opens a network connection.
     * This method is called before a test is executed.
     */
    protected function setUp()
    {
        $this->object = new Exception('test');
    }

    /**
     * @expectedException \Ease\Exception
     * @expectedExceptionMessage test
     */
    public function testConstructor()
    {
        $classname = get_class($this->object);

        // Get mock, without the constructor being called
        $mock = $this->getMockBuilder($classname)
            ->disableOriginalConstructor()
            ->getMockForAbstractClass();
        $mock->__construct('test');
        throw new Exception('test');
    }
    
    /**
     * Tears down the fixture, for example, closes a network connection.
     * This method is called after a test is executed.
     */
    protected function tearDown()
    {
        
    }
}
